src/components/nve-input/nve-input.component.ts

```diff
@@ -7,31 +7,98 @@ import styles from './nve-input.styles';
  * Mer info: https://shoelace.style/components/input
  *
  * Vil du ha info-ikon med hjelpetekst etter ledeteksten, putt en nve-label i label-slot.
- *
  * Disse attributtene skal ikke brukes:
  * - pill
  *
- * TODO: Utropstegn-ikon ved valideringsfeil
- * TODO: Vise valideringsfeil med rød tekst under tekstfeltet
+ * TODO: Felte blir breddere når feil ikone vises. Alt på grunn av at det dukker opp i en slot. Hvis Vi bestemmer oss
+ * å ha en fast verdi på sloten, kan det kanksje påvirke andre elementer som skal vises i sloten.
  */
 @customElement('nve-input')
 export default class NveInput extends SlInput {
   /**
    * Tekst som vises for å markere at et felt er obligatorisk. Er satt til "*Obligatorisk" som standard.
    */
   @property() requiredLabel = '*Obligatorisk';
+  /**
+   * Brukes til enkel constraint validation til å overskrive default nettleseren melding
+   */
+  @property({ reflect: true }) errorMessage?: string;
+  /**
+   * Hjelpe variabler som sjekker om input feltet er allerede invalid
+   */
+  @state() protected alreadyInvalid = false;
+
+  static styles = [SlInput.styles, styles];
 
   constructor() {
     super();
   }
 
-  static styles = [SlInput.styles, styles];
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener('sl-invalid', (e) => {
+      // vi vil ikke at nettleseren viser feil meldingen til oss
+      e.preventDefault();
+      if (!this.alreadyInvalid) {
+        this.makeInvalid();
+      }
+    });
+  }
 
-  updated(changedProperties: Map<string, unknown>) {
-    if (changedProperties.has('requiredLabel')) {
+  firstUpdated(): void {
+    if (this.requiredLabel) {
       this.style.setProperty('--sl-input-required-content', `"${this.requiredLabel}"`);
     }
   }
+
+  updated(): void {
+    if (this.hasAttribute('data-user-invalid') && !this.alreadyInvalid) {
+      this.makeInvalid();
+    }
+    if (!this.hasAttribute('data-user-invalid')) {
```

Comment:
I denne funksjonen kaller du hasAttribute() to ganger. Blir det kjappere om du gjør det kun en gang?